services/actions/notifier_helper.go

```diff
@@ -144,6 +146,22 @@ func notify(ctx context.Context, input *notifyInput) error {
 		return fmt.Errorf("gitRepo.GetCommit: %w", err)
 	}
 
+	skipRunEvents := []webhook_module.HookEventType{
+		webhook_module.HookEventPush,
+		webhook_module.HookEventPullRequest,
+		webhook_module.HookEventPullRequestSync,
+	}
+	if slices.Contains(skipRunEvents, input.Event) {
+		// skip runs with a configured skip-ci string in commit message if the event is push or pull_request(_sync)
+		// https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs
+		for _, s := range setting.Actions.SkipRunStrings {
+			if strings.Contains(commit.CommitMessage, s) {
+				log.Debug("repo %s with commit %s: skipped run because of %s string", input.Repo.RepoPath(), commit.ID, s)
+				return nil
+			}
+		}
+	}
```

Comment:
I recommend extracting this into a function called `skipWorkflowsForCommit` returning a boolean.